dotCMS/src/main/java/com/dotcms/browser/BrowserAPIImpl.java

```diff
@@ -0,0 +1,524 @@
+                browserQuery.offset, esSortBy, browserQuery.user, true);
+
+        for (final Contentlet contentlet : contentlets) {
+
+            Map<String, Object> contentMap = null;
+            if (contentlet.getBaseType().get() == BaseContentType.FILEASSET) {
+
+                final FileAsset fileAsset = APILocator.getFileAssetAPI().fromContentlet(contentlet);
+                contentMap = fileAssetMap(fileAsset, browserQuery.user, browserQuery.showArchived);
+            }
+
+            if (contentlet.getBaseType().get() == BaseContentType.DOTASSET) {
+
+                contentMap = dotAssetMap(contentlet, browserQuery.user, browserQuery.showArchived);
+            }
+
+            if (contentlet.getBaseType().get() == BaseContentType.HTMLPAGE) {
+
+                final HTMLPageAsset page = APILocator.getHTMLPageAssetAPI().fromContentlet(contentlet);
+                contentMap = htmlPageMap(page, browserQuery.user, browserQuery.showArchived, browserQuery.languageId);
+            }
+
+            final List<Integer> permissions = permissionAPI.getPermissionIdsFromRoles(contentlet, roles, browserQuery.user);
+            final WfData wfdata             = new WfData(contentlet, permissions, browserQuery.user, browserQuery.showArchived);
+            contentMap.put("wfActionMapList", wfdata.wfActionMapList);
+            contentMap.put("contentEditable", wfdata.contentEditable);
+            contentMap.put("permissions",     permissions);
+            returnList.add(contentMap);
+        }
+
+        // Filtering
+        returnList = this.filterReturnList(browserQuery, returnList);
+
+        // Sorting
+        Collections.sort(returnList, new WebAssetMapComparator(browserQuery.sortBy, browserQuery.sortByDesc));
+
+        int offset     = browserQuery.offset;
+        int maxResults = browserQuery.maxResults;
+        // Offsetting
+        if (offset < 0) {
+
+            offset = 0;
+        }
+
+        if (maxResults <= 0) {
+
+            maxResults = returnList.size() - offset;
+        }
+
+        if (maxResults + offset > returnList.size()) {
+
+            maxResults = returnList.size() - offset;
+        }
+
+        final Map<String, Object> returnMap = new HashMap<>();
+        returnMap.put("total", returnList.size());
+        returnMap.put("list",  returnList.subList(offset, offset + maxResults));
+        return returnMap;
+    }
+
+    private List<Map<String, Object>> filterReturnList(final BrowserQuery browserQuery, final List<Map<String, Object>> returnList) {
+
+        final List<Map<String, Object>> filteredList = new ArrayList<Map<String, Object>>();
+        for (final Map<String, Object> asset : returnList) {
+
+            String name = (String) asset.get("name");
+            name = name == null ? StringPool.BLANK : name;
+
+            String description = (String) asset.get("description");
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [The String literal "description" appears 5 times in this file; the first occurrence is on line 166](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5337824)